app/Actions/Review/GetReviewUsersDislikedAction.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+namespace Hedonist\Actions\Review;
+
+use Hedonist\Repositories\Review\ReviewRepositoryInterface;
+
+class GetReviewUsersDislikedAction
+{
+    private $reviewRepository;
+
+    public function __construct(ReviewRepositoryInterface $repository)
+    {
+        $this->reviewRepository = $repository;
+    }
+
+    public function execute(GetReviewUsersDislikedRequest $request): GetReviewUsersDislikedResponse
+    {
+        $dislikes = $this->reviewRepository->getById($request->getReviewId())->dislikes->all();
+        $users = [];
+        foreach ($dislikes as $dislike) {
+            $users[] = $dislike->user->info;
```

Comment:
no, this will be $dislike->user->info; db query in loop.

Use smth like DislikeRepository::findByCriteria(new ByReviewWithUsersCriteria(
 $reviewId
))

And inside
 $model
     ->where('review_id', $reviewId)
     ->with('users')